src/consumption.cpp

```diff
@@ -992,8 +992,9 @@ ret_val<edible_rating> Character::will_eat( const item &food, bool interactive )
         add_consequence( _( "Your stomach won't be happy (not rotten enough)." ), ALLERGY_WEAK );
     }
 
+    units::volume in_stomach_volume = food.volume() * compute_effective_food_volume_ratio( food );
```

Comment:
```suggestion
    units::volume in_stomach_volume = food.volume( false, false, 1 ) * compute_effective_food_volume_ratio( food );
```
This will solve the test failure. It makes the function not look at the faucet tank as a whole, but only see how much one charge would add. Previously, the test would not drink any water because seeing that it was 8 units (2L, 0.25L per) as a whole had the faucet thinking you were too full to safely consume water, choosing to become greyed-out (not `_enabled`) instead, causing `!stomach_should_be_full` to experience being empty while the faucet was disabled.

Reply:
I was a little worried about this because Character::consume variably returns a "All of the item was eaten" or "Some of the item was eaten" value, but it seems that distinction is only for if a charged item lost charges without hitting 0. Basically, we only ever eat one charge at a time, so calculating eaten volume based on 1 charge is objectively correct.

In simpler terms: looks good! 👍